netbox/utilities/filters.py

```diff
@@ -62,7 +61,13 @@ class TreeNodeMultipleChoiceFilter(django_filters.ModelMultipleChoiceFilter):
     """
     Filters for a set of Models, including all descendant models within a Tree.  Example: [<Region: R1>,<Region: R2>]
     """
+
     def filter(self, qs, value):
+        if settings.FILTERS_NULL_CHOICE_VALUE in value:
+            # Filtering by null value. Example: region=null
+            qs = self.get_method(qs)(**{self.field_name.replace('in', 'isnull'): True})
+            return qs.distinct() if self.distinct else qs
```

Comment:
This approach doesn't allow the user to specify `null` OR a specific region(s); e.g. `GET /api/dcim/sites/?region=us&region=null`. Can we extend this to modify the queryset to look like `Site.objects.filter(Q(region__isnull=True) | Q(region__in=[...]))`? Not sure how painful that is to do within the filter.

Reply:
Oh, I will try to find the way later. thx

Reply:
Applied to override `get_filter_predicate` for filtering 'null' value. This makes us possible to filter multiple region filters including 'null'. I also add tests to check this.